oak-auth-external/src/test/java/org/apache/jackrabbit/oak/spi/security/authentication/external/impl/principal/SyncHandlerMappingTrackerTest.java

```diff
@@ -72,23 +73,23 @@ public void testAddingServiceWithProperties() {
         when(ref.getProperty(PARAM_IDP_NAME)).thenReturn("testIDP");
 
         tracker.addingService(ref);
-        assertEquals(Set.of("testIDP"), ImmutableSet.copyOf(tracker.getIdpNames("testSH")));
+        assertEquals(Set.of("testIDP"), Collections.unmodifiableSet(CollectionUtils.toSet(tracker.getIdpNames("testSH"))));
 
         ServiceReference ref2 = mock(ServiceReference.class);
         when(ref2.getProperty(PARAM_SYNC_HANDLER_NAME)).thenReturn("testSH-2");
         when(ref2.getProperty(PARAM_IDP_NAME)).thenReturn("testIDP-2");
         tracker.addingService(ref2);
 
-        assertEquals(Set.of("testIDP"), ImmutableSet.copyOf(tracker.getIdpNames("testSH")));
-        assertEquals(Set.of("testIDP-2"), ImmutableSet.copyOf(tracker.getIdpNames("testSH-2")));
+        assertEquals(Set.of("testIDP"), Collections.unmodifiableSet(CollectionUtils.toSet(tracker.getIdpNames("testSH"))));
+        assertEquals(Set.of("testIDP-2"), Collections.unmodifiableSet(CollectionUtils.toSet(tracker.getIdpNames("testSH-2"))));
 
         ServiceReference ref3 = mock(ServiceReference.class);
         when(ref3.getProperty(PARAM_SYNC_HANDLER_NAME)).thenReturn("testSH");
         when(ref3.getProperty(PARAM_IDP_NAME)).thenReturn("testIDP-3");
         tracker.addingService(ref3);
 
-        assertEquals(Set.of("testIDP", "testIDP-3"), ImmutableSet.copyOf(tracker.getIdpNames("testSH")));
-        assertEquals(Set.of("testIDP-2"), ImmutableSet.copyOf(tracker.getIdpNames("testSH-2")));
+        assertEquals(Set.of("testIDP", "testIDP-3"), Collections.unmodifiableSet(CollectionUtils.toSet(tracker.getIdpNames("testSH"))));
```

Comment:
Do we need conversions to immutable when the set id only used in a subsequent use of "assert"? Maybe we can check the diff for uses of unmodifiableSet in the context of asserts?

Reply:
good catch @reschke. I would fix it.

Reply:
fixed in https://github.com/apache/jackrabbit-oak/pull/1914/commits/412cad0bd43ffc420df2ecde81ddbfadebeaeade along with other uses of `Collections.unmodifiableSet` in test cases.